src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSessionData.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion.Data;
+using Microsoft.VisualStudio.Text;
+using RoslynCompletionList = Microsoft.CodeAnalysis.Completion.CompletionList;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncCompletion
+{
+    /// <summary>
+    /// Contains data need to be tracked over an entire IAsyncCompletionSession completion
+    /// session for various operations.
+    /// </summary>
+    internal sealed class CompletionSessionData
+    {
+        private const string RoslynCompletionSessionData = nameof(RoslynCompletionSessionData);
+        public bool TargetTypeFilterExperimentEnabled { get; set; }
+        public bool TargetTypeFilterSelected { get; set; }
+        public bool HasSuggestionItemOptions { get; set; }
+
+        public Optional<SnapshotPoint> ExpandedItemTriggerLocation { get; set; }
+        public Optional<TextSpan> CompletionListSpan { get; set; }
```

Comment:
can all of these be optional at different times?  Or does one being provided mean the rest are provided as well?